nasa-gibs/worldview
# wv-2848: Shortlink update

## Description
If you keep Share link tab open, check shorten link, copy link, uncheck shorten link, move map/add layers/change date, & re-check "shorten link", the shortened link stays the same as the original link. In other words, the shortened link does not update after the initial shortening happens.

Fixes # wv-2848

## How to Reproduce
- Open Worldview
- Zoom into some feature
- Click the Share icon and then check "Shorten Link" 
- Make note of the link provided, then uncheck "Shorten Link" (keep Share window open!)
- Do something to update the URL permalink (move the map, add new layer, change the date, etc.)
- Click "Shorten Link" again
- Notice that the link is the same as the previous shortened link even though the URL parameters changed

## How To Test
Unfortunately, this is somewhat annoying to test locally because `web\js\containers\share.js` line 31 has a specific condition to prevent extra calls from localhost. To check if the **state** is updating, add logs in onToggleShorten():

`console.log('shortLinkKey', shortLinkKey);`
`console.log('queryString', queryString);`

Confirm that when you check the "Shorten Link" checkbox, the queryString value matches the current URL.

## Merging

Please use the `squash and merge` commit method unless each commit in your branch is vital to the commit history of main.

@nasa-gibs/worldview
